WarriorLibrary/BattleArena.cs

```diff
@@ -0,0 +1,31 @@
+namespace WarriorLibrary;
+
+public static class BattleArena
+{
+    public static int roundCounter { get; set; }
+    public static List<BaseWarrior> Duel(BaseWarrior firstBaseWarrior, BaseWarrior secondBaseWarrior)
+    {
+        roundCounter = 0;
+        for (int i = 0; firstBaseWarrior.HasAlive && secondBaseWarrior.HasAlive; i++)
+        {
+            ++roundCounter;
+            //Console.WriteLine($"\tRound {++roundCounter}\n{firstBaseWarrior.ToString()}\n{secondBaseWarrior.ToString()}");
+            if (i % 2 == 0)
+                secondBaseWarrior.Defence(firstBaseWarrior.AttackPower());
+            else
+                firstBaseWarrior.Defence(secondBaseWarrior.AttackPower());
+        }
+
+        var winner = firstBaseWarrior.HasAlive
+            ? firstBaseWarrior
+            : secondBaseWarrior.HasAlive
+                ? secondBaseWarrior
+                : null;
+        var looser = winner != firstBaseWarrior
+            ? firstBaseWarrior
+            : winner == secondBaseWarrior
+                ? secondBaseWarrior
+                : null;
+        return new List<BaseWarrior>{winner!, looser!};
```

Comment:
It is not the best solution.  1. You can have a situation when you'll have no winner and no looser.
2. Why do you need to return the list which contains only 2 elements every time? You have the same data by references in the class which calls this method. The order in the list is not clear that the first one in the list is the winner.